staker/rollup_watcher.go

```diff
@@ -73,15 +77,41 @@ func (r *RollupWatcher) getCallOpts(ctx context.Context) *bind.CallOpts {
 	return &opts
 }
 
+const noNodeErr string = "NO_NODE"
+
+func looksLikeNoNodeError(err error) bool {
+	if err == nil {
+		return false
+	}
+	if strings.Contains(err.Error(), noNodeErr) {
+		return true
+	}
+	var errWithData rpc.DataError
+	ok := errors.As(err, &errWithData)
+	if !ok {
+		return false
+	}
+	dataString, ok := errWithData.ErrorData().(string)
+	if !ok {
+		return false
+	}
+	data := common.FromHex(dataString)
+	return bytes.Contains(data, []byte(noNodeErr))
+}
+
 func (r *RollupWatcher) getNodeCreationBlock(ctx context.Context, nodeNum uint64) (*big.Int, error) {
 	callOpts := r.getCallOpts(ctx)
 	if !r.unSupportedL3Method.Load() {
 		createdAtBlock, err := r.GetNodeCreationBlockForLogLookup(callOpts, nodeNum)
 		if err == nil {
+			r.supportedL3Method.Store(true)
 			return createdAtBlock, nil
 		}
-		log.Trace("failed to call getNodeCreationBlockForLogLookup, falling back on node CreatedAtBlock field", "err", err)
-		if headerreader.ExecutionRevertedRegexp.MatchString(err.Error()) {
+		if headerreader.ExecutionRevertedRegexp.MatchString(err.Error()) && !looksLikeNoNodeError(err) {
+			if r.supportedL3Method.Load() {
+				return nil, fmt.Errorf("getNodeCreationBlockForLogLookup failed despite previously succeeding: %w", err)
+			}
+			log.Trace("failed to call getNodeCreationBlockForLogLookup, falling back on node CreatedAtBlock field", "err", err)
```

Comment:
Should be at least Info - since this will be printed exactly once, and could help debugging some weird cases.

Reply:
Makes sense. I'm a bit hesitant in general to make stuff like this "info" because 3rd party teams will complain about the "issue" in their logs, or think it's the cause of some other issue, but it indeed might be useful for debugging. I'll upgrade it to info, and maybe think about rewording it.

Reply:
Upgraded to info and reworded a bit to help clarify it for 3rd party teams